### DISCUSSION TITLE
Ranger's rename file workflow?

### DISCUSSION THREAD

**Discussion by pablo-888** [sxyazi/yazi#292] [1. Q&A]
Is there a way to recreate that currently?. In Ranger you can press 'a' to open the rename dialogue which automatically places the cursor before the file extension + using ctrl-w to delete entire words rather than having to hold backspace which is way slower if the filename is too long.

Thanks!.

**Comment by sxyazi** [comment]
I don't quite understand what you mean. AFAIK, Ctrl-W only deletes the current word. So are you trying to delete a word within the file name(without ext), or the entire file name(without ext)? Could you provide an example for it?

**Reply by pablo-888** [reply]
> AFAIK, Ctrl-W only deletes the current word.
 
it does? Ctrl-W does nothing at all for me on the rename dialogue inside yazi so I assumed that functionality just wasn't there. Maybe there's a keybinding i'm missing on keymap.toml?

**Reply by sxyazi** [reply]
Um, I meant ranger. I want to know your specific use case, whether you need to delete the entire filename or just one word from it?

**Reply by pablo-888** [reply]
Oh, sorry. Yeah I just meant deleting entire words. (like in Bash when you do ctrl-w basically).

**Reply by sxyazi** [reply]
Yazi's input is a standalone Vi-like input component that allows combining multiple commands. I'm wondering how you would edit this format in Vim. Maybe I can add a command to jump to before the extension? Not sure if there's a better way to achieve it.

**Comment by sxyazi** [comment]
Supported in https://github.com/sxyazi/yazi/pull/513, doc https://yazi-rs.github.io/docs/configuration/keymap#manager.rename
